internal/bminventory/inventory.go

```diff
@@ -392,6 +398,68 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 	return installer.NewRegisterClusterCreated().WithPayload(&cluster.Cluster)
 }
 
+func (b *bareMetalInventory) RegisterAddHostsCluster(ctx context.Context, params installer.RegisterAddHostsClusterParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	id := params.NewAddHostsClusterParams.ID
+	url := installer.GetClusterURL{ClusterID: *id}
+	consoleURL := swag.StringValue(params.NewAddHostsClusterParams.APIVipDnsname)
+	clusterName := swag.StringValue(params.NewAddHostsClusterParams.Name)
+	openshiftVersion := swag.StringValue(params.NewAddHostsClusterParams.OpenshiftVersion)
+
+	log.Infof("Register add-hosts-cluster: %s with id %s", clusterName, id.String())
+
+	cluster := common.Cluster{Cluster: models.Cluster{
+		ID:               id,
+		Href:             swag.String(url.String()),
+		Kind:             swag.String(models.ClusterKindAddHostsCluster),
+		Name:             clusterName,
+		OpenshiftVersion: openshiftVersion,
+		UpdatedAt:        strfmt.DateTime{},
+	}}
+
+	err := validations.ValidateClusterNameFormat(clusterName)
+	if err != nil {
+		return common.NewApiError(http.StatusBadRequest, err)
+	}
+
+	// Persist worker-ignition to s3 for cluster
+	ignitionConfig, err := b.formatNodeIgnitionFile(consoleURL)
+	if err != nil {
+		log.WithError(err).Errorf("failed to format ignition config file for cluster %s", cluster.ID)
+	}
+	fileName := fmt.Sprintf("%s/%s", cluster.ID, workerIgnition)
+	err = b.objectHandler.Upload(ctx, ignitionConfig, fileName)
+	if err != nil {
+		return installer.NewRegisterAddHostsClusterInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, fmt.Errorf("failed to upload %s to s3", fileName)))
+	}
+
+	// After registering the cluster, its status should be 'ClusterStatusAddingHosts'
+	err = b.clusterApi.RegisterAddHostsCluster(ctx, &cluster)
+	if err != nil {
+		log.Errorf("failed to register cluster %s ", clusterName)
+		return installer.NewRegisterAddHostsClusterInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+
+	return installer.NewRegisterAddHostsClusterCreated().WithPayload(&cluster.Cluster)
+}
+
+func (b *bareMetalInventory) formatNodeIgnitionFile(apiVipDnsname string) ([]byte, error) {
+	var ignitionParams = map[string]string{
+		"SOURCE": "http://" + apiVipDnsname + ":22624/config/worker",
```

Comment:
Will this ever be https?  Or different port?

Reply:
no, we will always use http since we have no way to get certificate from OCP cluster, and the MCO http port won't change soon